tools/xfsslower.py

```diff
@@ -187,10 +187,8 @@
     bpf_get_current_comm(&data.task, sizeof(data.task));
 
     // workaround (rewriter should handle file to d_name in one step):
-    struct dentry *de = NULL;
-    struct qstr qs = {};
-    bpf_probe_read(&de, sizeof(de), &valp->fp->f_path.dentry);
-    bpf_probe_read(&qs, sizeof(qs), (void *)&de->d_name);
+    struct dentry *de = valp->fp->f_path.dentry;
+    struct qstr qs = de->d_name;
```

Comment:
You can do just this:

```
struct qstr qs = valp->fp->f_path.dentry->d_name;
```

Reply:
Oh, yes. Looks like we can :-)
That addresses the above comment too. Thanks!